### PR TITLE
[GFX11] Appropriate GFX10 workarounds and fixes reused for the new target. [GFX11][tests] Tests enabled. 

### PR DESCRIPTION
**Pull request by atamazov**
- Appropriate GFX10 workarounds extended to cover GFX11:
  - `WORKAROUND_SWDEV_271887`
  - `WORKAROUND_SWDEV_266868`
  - `WORKAROUND_MIOPEN_ISSUE_557` (see #557) 
  - `WORKAROUND_ISSUE_1053` (see #1053)
  - `WORKAROUND_ISSUE_1148` (see #1148)
  - `WORKAROUND_ISSUE_1431` (see #1431)
- [COMGR] Selected wavefrontsize32 device libs when necessary
- [tests]
  - Added support for detection of gfx1100 and gfx1102 as well as manual selection (`-DMIOPEN_TEST_GFX110X=On`)
    - Borrowed from _[GFX11][tests] New target additional list file changes #1923_ by @muralinr 
  - [OCL BE] Skiped the same tests for GFX11 as we already skip for GFX10
  - Added support for the GFX103X_ENABLED/DISABLED attribute
  - Enabled appropriate custom tests (GFX103X_ENABLED)
- By-products
  - [tests][NFC] Reworked docs: Changed "options" to "attributes". Removed useless text.

@junliume @muralinr Recommended labels: 
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/enhancement
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/urgency_high (am I right?)
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/testing
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/workaround

### CODE REVIEW

**Review by muralinr — COMMENTED** [review]
Changes look good.

**Comment by muralinr** [comment]
I will test this change after batchnorm, convolution, and other gfx11xx changes PRs are merged.

**Comment by junliume** [comment]
@atamazov  @muralinr  is this one replacing #1923 and #1924 ?

**Comment by muralinr** [comment]
> @atamazov @muralinr is this one replacing #1923 and #1924 ?

No, It only replaces #1923.  We still need #1924.

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@muralinr How fast we are! Does it work without problems on GFX11? That would be good news ;)

/cc @junliume

**Comment by muralinr** [comment]
> @muralinr How fast we are! Does it work without problems on GFX11? That would be good news ;)
> 
> /cc @junliume

I am running tests on GFX11. No issues on GFX11 testing so far.

**Comment by muralinr** [comment]
> > @muralinr How fast we are! Does it work without problems on GFX11? That would be good news ;)
> > /cc @junliume
> 
> I am running tests on GFX11. No issues on GFX11 testing so far.

Most of tests are running fine. No issues on GFX11 testing so far. I will continue to run remaining tests.

**Comment by atamazov** [comment]
@muralinr Good! Are we going to enable HIP Implicit Gemm solvers for GFX11 soon?

**Comment by muralinr** [comment]
> > > @muralinr How fast we are! Does it work without problems on GFX11? That would be good news ;)
> > > /cc @junliume
> > 
> > 
> > I am running tests on GFX11. No issues on GFX11 testing so far.
> 
> Most of tests are running fine. No issues on GFX11 testing so far. I will continue to run remaining tests.

I don't see any test failures.  All tests are fine

**Comment by muralinr** [comment]
> @muralinr Good! Are we going to enable HIP Implicit Gemm solvers for GFX11 soon?

We are currently mainlining all Navi3.x fixes.  I will let @junliume to comment on implicit GEMM solvers.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -100,6 +101,10 @@ if(NOT (MIOPEN_TEST_GFX900 OR MIOPEN_TEST_GFX906 OR MIOPEN_TEST_GFX908 OR MIOPEN
             set(MIOPEN_TEST_GPU_DETECTION_FAILED TRUE)
         elseif(ROCMINFO_OUTPUT MATCHES "gfx1030")
             set(MIOPEN_TEST_GFX103X ON)
+        elseif(ROCMINFO_OUTPUT MATCHES "gfx1100")
+            set(MIOPEN_TEST_GFX110X ON)
+        elseif(ROCMINFO_OUTPUT MATCHES "gfx1102")
```
Comment (atamazov) [inline test/CMakeLists.txt:106]: ⚓ +gfx1101?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -38,6 +38,7 @@ option( MIOPEN_TEST_GFX90A "Test on gfx90a" OFF )
 option( MIOPEN_TEST_GFX900 "Test on Vega10 (gfx900)" OFF )
 option( MIOPEN_TEST_GFX906 "Test on Vega20 (gfx906)" OFF )
 option( MIOPEN_TEST_GFX103X "Test on Navi21/22 (gfx1030/31)" OFF )
+option( MIOPEN_TEST_GFX110X "Test on Navi31/32 (gfx1100/02)" OFF )
```
Comment (atamazov) [inline test/CMakeLists.txt:41]: ⚓ "...Navi31/32/33 (gfx1100/02/01)" ?

**Pull request merged**